File: test/perf2/laplace/__laplace.c
void c_update(double **u, unsigned long rows, unsigned long cols, double dx2, double dy2) {
    unsigned long i, j;
    for (i=1; i<rows-1; i++)
        for (j=1; j<cols-1; j++)
            u[i][j] = ((u[i+1][ j] + u[i-1][ j]) * dy2 + (u[i][ j+1] + u[i][ j-1]) * dx2) * (1.0 / (2*(dx2+dy2)));
}

File: test/perf2/laplace/laplace.m
% Run this script with tic; laplace; toc, and multiply the time by
% 1000 to get time in ms - in order to compare with laplace.jl.

dx = 0.1;
dy = 0.1;
dx2 = dx*dx;
dy2 = dy*dy;

N     = 150;
Niter = 2^10;
 
u = zeros(N,N);
u(1,:) = 1;

for i = 1:Niter
    u(2:end-1, 2:end-1) = ((u(1:end-2, 2:end-1) + u(3:end, 2:end-1))*dy2 + (u(2:end-1,1:end-2) + u(2:end-1, 3:end))*dx2) * (1./ (2*(dx2+dy2)));
end
